smm/protocols/__init__.py

```diff
@@ -22,8 +23,14 @@ def __init__(self, ql):
         self.comm_buffer_size = ql.os.heap.alloc(ctypes.sizeof(ctypes.c_void_p))
         ql.mem.write(self.comm_buffer_size, ctypes.sizeof(EFI_SMM_SW_CONTEXT).to_bytes(ctypes.sizeof(ctypes.c_void_p), 'little'))
         
-        # Reserve SMRAM
-        ql.mem.map(self.smbase, self.smram_size)
+        if ql.mem.is_available(self.smbase, self.smram_size):
+            heap_size = 0x10000
```

Comment:
If we are being pedantic (and we are :) smm_heap_size should probably be defined by the ini.